src/NuGet.Clients/PackageManagement.VisualStudio/VSNuGetProjectFactory.cs

```diff
@@ -54,13 +54,16 @@ public NuGetProject CreateNuGetProject(EnvDTEProject envDTEProject, INuGetProjec
 
             NuGetProject result = null;
 
+            var projectId = VsHierarchyUtility.GetProjectId(envDTEProject);
```

Comment:
We're already doing this as part of our project telemetry events... I haven't added anything new just moved it to right place so that it could be used across nuget project